File: src/my_cpp_pkg/src/number_counter.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int64.hpp"

class numberCounterNode : public rclcpp::Node
{
public:
    numberCounterNode() : Node("number_counter"), publish_counted_number_(0)
    {
        publisher_ = this->create_publisher<example_interfaces::msg::Int64>("number_counter", 10);
        subscriber_ = this->create_subscription<example_interfaces::msg::Int64>(
            "number", 10,
            std::bind(&numberCounterNode::callbackCounter, this, std::placeholders::_1));

        RCLCPP_INFO(this->get_logger(), "Number Counter Node Started.");
    }

private:
    void callbackCounter(const example_interfaces::msg::Int64::SharedPtr msg)
    {
        publish_counted_number_ += msg->data;
        auto new_msg = example_interfaces::msg::Int64();
        new_msg.data = publish_counted_number_;
        publisher_->publish(new_msg);
        RCLCPP_INFO(this->get_logger(), "%d", publish_counted_number_);
    }

    int publish_counted_number_;
    rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr publisher_;
    rclcpp::Subscription<example_interfaces::msg::Int64>::SharedPtr subscriber_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<numberCounterNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: ros2_ws/src/my_py_pkg/my_py_pkg/number_counter.py
#!/usr/bin/env python3
import rclpy
from rclpy.callback_groups import CallbackGroup
from rclpy.node import Node

from example_interfaces.msg import Int64


class numberCounterNode(Node):
    def __init__(self):
        super().__init__("number_counter")
        self.counter_ = 0
        self.subscriber_ = self.create_subscription(
            Int64, "number_publish", self.callback_number_count, 10
        )
        self.get_logger().info("Counter node Startted")

    def callback_number_count(self, msg):
        self.counter_ += msg.data
        self.get_logger().info(str(self.counter_))


def main(args=None):
    rclpy.init(args=args)
    node = numberCounterNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()
